pkg/controller/v1beta1/inferenceservice/controller.go

```diff
@@ -156,8 +161,10 @@ func (r *InferenceServiceReconciler) Reconcile(ctx context.Context, req ctrl.Req
 			}
 
 			// remove our finalizer from the list and update it.
-			isvc.ObjectMeta.Finalizers = utils.RemoveString(isvc.ObjectMeta.Finalizers, finalizerName)
-			if err := r.Update(context.Background(), isvc); err != nil {
+			controllerutil.RemoveFinalizer(isvc, finalizerName)
+			patchYaml := "metadata:\n  finalizers: [" + strings.Join(isvc.ObjectMeta.Finalizers, ",") + "]"
+			patchJson, _ := yaml.YAMLToJSON([]byte(patchYaml))
+			if err := r.Patch(ctx, isvc, client.RawPatch(types.MergePatchType, patchJson)); err != nil {
```

Comment:
Anything needs to be changed in https://github.com/kserve/kserve/tree/master/config/rbac?

Reply:
I guess no.  It already has `patch` privilege. https://github.com/kserve/kserve/blob/master/config/rbac/role.yaml#L141-L153